### ISSUE TITLE
It's so painful to do an update on a One to Many relationship

### ISSUE THREAD

**Issue by pedro-areal-torres** [typeorm/typeorm#10553]
### Issue description

Can't update 

### Expected Behavior

Using as an example something simple:
- A user can have many departments
- Departments is an enum
- For each department a user needs to have a desk number 

User Entity:
```
@Entity()
export class User extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @OneToMany(() => UserDepartment, (department) => department.user, {
    cascade: true,
    orphanedRowAction: 'delete', // comments about this below
  })  departments: UserDepartment[];
}

```

Department Entity:
```
@Entity('user_department')
@Unique(['userId', 'department']) 
export class FabricComposition extends BaseEntity {
  @PrimaryColumn({ name: 'user_id' })
  userId: number;

  @ManyToOne(() => User)
  @JoinColumn({ name: 'user_id' })
  user: User;

  @PrimaryColumn({ type: 'enum', enum: DepartmentType })
  department: DepartmentType;

  @Column()
  desk: number;
}
```

If I want to create a user, it's quite easy:
Request:
```
{
 "name": "John",
 "departments": [
      {
        "department": ABC,
        "desk": 1
      },
      {
        "department": DEF,
        "desk": 1
      },
  ] 
}
```

Service: 
```
this.userRepository.save({ createUserDto });
```

NOW, if I want to update, then I have no idea how to avoid 50 lines to make this work.
Request:
```
{
 "name": "John",
 "departments": [
      {
        "department": ABC,
        "desk": 2 // Updated desk number
      },
      {
        "department": XYZ, // Switched from department DEF to XYZ
        "desk": 3
      },
  ] 
}
```
Service:
If I try:
this.userRepository.update({ id }, updateUserDto);
I will get as an error:
Cannot query across one-to-many for property composition

What I need is:
- Update the details on existing department
- Delete the department that user doesn't belong anymore
- Add the new department to the user

The only way that I've found was doing like many loops (one to find the department to delete, then one to save the departments one by one).

I really don't understand how does this works.

### Actual Behavior

Getting error:
`Cannot query across one-to-many for property composition`

### Steps to reproduce

Mentioned on "Expect behavior".

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Node.js version     | 18.17.0  |
| Typescript version  | 4.7.4 |
| TypeORM version     | 0.3.9 |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

**Comment by MonstraG** [comment]
Yep, still a problem. Doing orphanedRowAction: delete will "work" by deleting all the rows, which is really less then desirable

**Comment by gioboa** [comment]
Thanks @MonstraG for the heads up, may I ask you to drop a PR for that?

**Comment by MonstraG** [comment]
I'm not the original poster, but I have also never looked inside of typeorm and this issue feels like isn't one, where somebody can just pass by, drop a pr, and not break lots of magical save-handling code.

**Comment by pedro-areal-torres** [comment]
This is a bit limitation that we still have to handle and workaround when we want to update multiple records that have OneToMany relationships.

As @MonstraG said, I guess that this is something that the new leadership has to take into account.
